exportReport/ExtensionExportReport.java

```diff
@@ -0,0 +1,298 @@
+ * CLASS		:	ExtensionExportReport.java 
+ * DESC			:	Initializes all data and creates a hook point to the report menu for the extension.
+ * CREATED ON	:	MARCH 10TH, 2016
+ * CURRENT VER	:	V1.0
+ * SOURCE		:	https://github.com/JordanGS/workspace/tree/master/zap-extensions/src/org/zaproxy/zap/extension/exportReport
+ */
+
+/* 
+ * MODIFED BY	:	<NAME> - <GIT USER>
+ * MOD DATE		:	
+ * MOD VERSION	:	<VERSION OF PLUGIN>
+ * MOD DESC		:	
+ */
+
+public class ExtensionExportReport extends ExtensionAdaptor
+{
+
+	public static final String			NAME				= "ExtensionExportReport";
+	private ZapMenuItem					menuExportReport	= null;
+	private FrameExportReport			frameER				= null;
+
+	protected final Map<String, String>	alertTypeRisk		= new HashMap<>();
+
+	/* Nav Panel creation. */
+	private PanelSource					cardSource			= null;
+	private PanelAlertRisk				cardAlertRisk		= null;
+	private PanelAlertDetails			cardAlertDetails	= null;
+
+	/* Field Limit Constants */
+	private final int					textfieldLimit		= 30;
+	private final int					textareaLimit		= 600;
+
+	/* Relational CONSTANT for export types. */
+	private ArrayList<String>			alertSeverity		= new ArrayList<String>();
+	private ArrayList<String>			alertDetails		= new ArrayList<String>();
+	private ArrayList<String>			alertAdditional		= new ArrayList<String>();
+	private int							maxList				= 0;
+	
+	public ExtensionExportReport()
+	{
+		super();
+		initialize();
+	}
+
+	public ExtensionExportReport(String name)
+	{
+		super(name);
+	}
+
+	private void initialize()
+	{
+		this.setName("ExtensionExportReport");
+	}
+
+	@Override
+	public void hook(ExtensionHook extensionHook)
+	{
+		super.hook(extensionHook);
+		if (getView() != null)
+		{
+			extensionHook.getHookMenu().addReportMenuItem(getMenuExportReport());
+		}
+	}
+
+	@Override
+	public boolean canUnload()
+	{
+		return true;
+	}
+
+	@Override
+	public void unload()
+	{
+		super.unload();
+		if (frameER != null)
+		{
+			frameER.dispose();
+			frameER = null;
+		}
+	}
+
+	private ZapMenuItem getMenuExportReport()
+	{
+		if (menuExportReport == null)
+		{
+			menuExportReport = new ZapMenuItem("menu.report.html.generate");
+			menuExportReport.setText(Constant.messages.getString("exportReport.menu.export"));
+			menuExportReport.addActionListener(new java.awt.event.ActionListener()
+			{
+				@Override
+				public void actionPerformed(java.awt.event.ActionEvent e)
+				{
+					if (frameER != null)
+					{
+						frameER.requestFocusInWindow();
+						return;
+					}
+					getNewOptionFrame();
+					
+					// -Change the default close operation
+					WindowAdapter exitListener = new WindowAdapter() {
+					    @Override
+					    public void windowClosing(WindowEvent e) {
+					    	emitFrame();
+					    }
+					};
+					frameER.addWindowListener(exitListener);
+					frameER.setSize(400, 500);
+					frameER.setResizable(false);
+					frameER.setVisible(true);
+					frameER.centerFrame();//frameER.setLocationRelativeTo(null);
```

Comment:
Well this is just an initial release as i am still hard at work adding other forms of export and some more features. They are currently useful for me in terms of debugging but if it will help others i am unsure but i was planning on removing all of those debug comments once i got my PDF export to work properly and pushed it for release as well.

As for comments such as `changes the default close operation`, i thought this was a valid addition because i had some issues with the tool not starting on a second run because `emitFrame` was not used to override the default close operation. I kept it in there as a reference at the time but i can remove it if you don't think it's useful or constructive.